### PR TITLE
Python: Update readme

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Development is made easy using [Poetry](https://python-poetry.org/docs/#installa
 
 ```bash
 poetry shell
-poetry install -E pyarrow
+poetry install -E pyarrow -E hive -E s3fs
```
Comment (nastra) [inline python/README.md:38]: tests don't pass without `s3fs` and so I figured we can align this with the make file, which uses `poetry install -E pyarrow -E hive -E s3fs`

Comment (Fokko) [reply python/README.md:38]: Thanks @nastra, great catch! It seems that this is outdated. What do you think of:
```sh
make install
make test
```
This way we just have to maintain the `Makefile`, that is also being used in the CI.
Makefile: https://github.com/apache/iceberg/blob/master/python/Makefile#L18-L20

Comment (nastra) [reply python/README.md]: makes sense, updated

**Review by Fokko — APPROVED** [review]
Thanks!

**Pull request merged**